### PR TITLE
a sprinkle of unicode

### PR DESCRIPTION
**Pull request by mr-c**

### CODE REVIEW

File: stdlib/2.7/logging/__init__.pyi [hunk 1]
```diff
@@ -170,13 +170,13 @@ class Logger(Filterer):
...
-    def warn(self, msg: str, *args, **kwargs) -> None: ...
-    def error(self, msg: str, *args, **kwargs) -> None: ...
-    def exception(self, msg: str, *args, **kwargs) -> None: ...
-    def critical(self, msg: str, *args, **kwargs) -> None: ...
+    def debug(self, msg: Union[str, unicode], *args, **kwargs) -> None: ...
+    def info(self, msg: Union[str, unicode], *args, **kwargs) -> None: ...
+    def warning(self, msg: Union[str, unicode], *args, **kwargs) -> None: ...
+    def warn(self, msg: Union[str, unicode], *args, **kwargs) -> None: ...
+    def error(self, msg: Union[str, unicode], *args, **kwargs) -> None: ...
+    def exception(self, msg: Unions[str, unicode], *args, **kwargs) -> None: ...
```
Comment (gvanrossum) [inline stdlib/2.7/logging/__init__.pyi]: Typo (Unions).

**Comment by gvanrossum** [comment]
Actually this would probably work just as well with just `unicode` instead of `Union[str, unicode]`, since mypy considers `str` a subclass of `unicode`.

**Comment by mr-c** [comment]
@gvanrossum Fixed, now just `unicode` & I added a couple more.

**Comment by matthiaskramm** [comment]
I'd prefer if we did not make the assumption that all type-checkers consider `str` a subclass of `unicode`.

I know mypy is one of the primary users of typeshed, but implementation details of mypy still shouldn't leak into typeshed.

**Comment by gvanrossum** [comment]
Really sorry, I had forgotten about that. Let's go back to the Unions, at least until we have agreement on what to do for strings (https://github.com/python/mypy/issues/1141 -- though we should really have the discussion in https://github.com/python/typing/issues).

**Comment by matthiaskramm** [comment]
Added https://github.com/python/typing/issues/204, in an attempt to merge the two discussion threads.

Yes, let's use Unions for the time being. Maybe it would make sense to have a "Text" type in typeshed only.

**Comment by mr-c** [comment]
Thank you all; this is ready for merging.

**Comment by gvanrossum** [comment]
Thanks!

**Pull request merged**